src/hw_isolation_event/hw_status_manager.cpp

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "config.h"
+
+#include "hw_isolation_event/hw_status_manager.hpp"
+
+#include <fmt/format.h>
+
+#include <phosphor-logging/elog-errors.hpp>
+
+#include <filesystem>
+
+namespace hw_isolation
+{
+namespace event
+{
+namespace hw_status
+{
+
+using namespace phosphor::logging;
+namespace fs = std::filesystem;
+
+constexpr auto HW_STATUS_EVENTS_PATH =
+    HW_ISOLATION_OBJPATH "/events/hw_isolation_status";
+
+Manager::Manager(sdbusplus::bus::bus& bus) : _bus(bus), _lastEventId(0)
+{}
+
+std::optional<sdbusplus::message::object_path> Manager::createEvent(
+    const EventSeverity& eventSeverity, const EventMsg& eventMsg,
+    const std::string& hwInventoryPath, const std::string& bmcErrorLogPath)
+{
+    try
+    {
+        auto id = _lastEventId + 1;
+        auto eventObjPath =
+            fs::path(HW_STATUS_EVENTS_PATH) / std::to_string(id);
+
+        // Add association for the hareware inventory path which needs
+        // the hardware status event.
+        // Note: Association forward and reverse type are defined as per
+        // xyz::openbmc_project::Logging::Event interface associations
+        // documentation.
+        type::AsscDefFwdType eventIndicatorFwdType{"event_indicator"};
+        type::AsscDefRevType eventIndicatorRevType{"event_log"};
+        type::AssociationDef associationDeftoEvent;
+        associationDeftoEvent.push_back(std::make_tuple(
+            eventIndicatorFwdType, eventIndicatorRevType, hwInventoryPath));
+
+        // Add the error_log if given
+        if (!bmcErrorLogPath.empty())
+        {
+            type::AsscDefFwdType errorLogFwdType{"error_log"};
+            type::AsscDefFwdType errorLogRevType{"event_log"};
+            associationDeftoEvent.push_back(std::make_tuple(
+                errorLogFwdType, errorLogRevType, bmcErrorLogPath));
+        }
+
+        _hwStatusEvents.insert(
+            std::make_pair(id, std::make_unique<hw_isolation::event::Event>(
+                                   _bus, eventObjPath, id, eventSeverity,
+                                   eventMsg, associationDeftoEvent)));
+
+        // Update the last event id using the created event id;
+        _lastEventId = id;
+        return eventObjPath.string();
+    }
+    catch (const std::exception& e)
+    {
+        log<level::ERR>(
```

Comment:
Journal is sufficient  here ? d-bus users creates error log with failure reason?

Reply:
We are not introducing any DBus API to create the event for the hardware deconfiguration reason and only supporting to pull the reason from the cec device tree if failed to boot the host or host reached to standby) so no action is required at the client-side, right?

are you seeing any use case where the user wants to update the reason? 

Even if we have a use case means we can handle it at the caller who called the `createEvent()` API by using the return type i.e is it `std::nullopt` or not.

Reply:
Are you providing enough data to  use to create the PEL ? I think the direction should be  create PEL with all required info , icluding callouts and return back to user. In general direction  should be failure case required error log , aznd we acn alway's trush journal log due to size limiation, most of the time , log data won't be availbe.

Reply:
> Are you providing enough data to use to create the PEL ? I think the direction should be create PEL with all required info , icluding callouts and return back to user.

I think, In this path, we are pulling the deconfiguration reasons from the guard records or the cec device tree and that reasons will be updated by the other application (BMC/Host applications) so those applications should take care of the callout info and PEL creation, right?

I think the openpower-hw-isolation application is not required to do any callout during guard record creation and event creation (for the deconfig reason) right?

> In general direction should be failure case required error log , aznd we acn alway's trush journal log due to size limiation, most of the time , log data won't be available.

Agree, [Created the BMC error log](https://github.com/ibm-openbmc/openpower-hw-isolation/pull/7/commits/2080d60f3b7ed933e12dc2a2645cb9463be375cb#diff-451290b7aa811e66f87b0c6357d76c6e6a3364cc6b4ad29462929080900a0034R291) with BMC dump at the caller place as per your other comments and also tested with the BMC dump as mentioned below in the different path so we will have the PEL and Dump for the future debug right?

```
grep "MESSAGE\|PRIORITY\|_SYSTEMD_UNIT" obmcdump_81_1638175859/journal-pid-3617.log | grep "PRIORITY=3" -A1 -A2

PRIORITY=3
 _SYSTEMD_UNIT=org.open_power.HardwareIsolation.service
MESSAGE=Unsupported hardware isolation entry [xyz.openbmc_project.HardwareIsolation.Entry.Type.Manual]
```